src/main/java/com/datalogics/pdf/samples/forms/FillForm.java

```diff
@@ -127,38 +90,10 @@ public static void main(final String... args) throws Exception {
      * @param outputUrl The file to which the filled form will be saved
      * @throws Exception a general exception was thrown
      */
-    public static void fillPdfForm(final URL inputFormUrl, final URL inputDataUrl, final String formType,
-                                   final URL outputUrl)
+    public static void fillPdfForm(final URL inputFormUrl, final URL inputDataUrl,
+                                   final FormImporter.FormType formType, final URL outputUrl)
                     throws Exception {
-        final PDFDocument pdfDocument = DocumentUtils.openPdfDocument(inputFormUrl);
-
-        // There are two types of forms that we can fill, so find out which kind we have here.
-        final PDFDocumentType documentType = XFAService.getDocumentType(pdfDocument);
-
-        if (documentType == PDFDocumentType.Acroform) {
-            // If this is an Acroform, make sure the form data is either FDF for XFDF.
-            if (FDF_FORMAT.equalsIgnoreCase(formType)) {
-                fillAcroformFdf(pdfDocument, inputDataUrl, outputUrl);
-            } else if (XFDF_FORMAT.equalsIgnoreCase(formType)) {
-                fillAcroformXfdf(pdfDocument, inputDataUrl, outputUrl);
-            } else {
-                throw new IllegalArgumentException("Invalid formData type for Acroform document. "
-                                                   + "FDF and XFDF supported.");
-            }
-        } else if (documentType.isXFA()) {
-            // If the document has an XFA form, make sure that we were passed an XML data file.
-            // Note that PDF Java Toolkit doesn't support generating appearances or running calculations on XFA forms
-            // (though field formatting is supported), so be sure to use Acrobat or another full-featured PDF viewer
-            // to verify the output. A viewer like OSX's Preview won't cut it.
-            if (XML_FORMAT.equalsIgnoreCase(formType)) {
-                fillXfa(pdfDocument, inputDataUrl, outputUrl);
-            } else {
-                throw new IllegalArgumentException("Invalid formData type for XFA document. XML supported.");
-            }
-        } else {
-            // If the form document is not XFA or Acroform, it's either not a form or it's something we don't support.
-            throw new IllegalArgumentException("PDF File does not contain an AcroForm or an XFA form.");
-        }
+        FormImporter.fillPdfForm(inputFormUrl, inputDataUrl, formType, outputUrl);
```

Comment:
I've been having a debate with myself about whether our samples should be designed as things that can be reused by other code. I tend to lean on the idea that their only value should be pedagogical—essentially, they exist only to be studied for their ideas and then 'thrown away'. (I realize my attitude may not be shared by the team.)

This PR is not the time to have that debate, however. I understand the reasons for leaving them in (even if I'm not sure I agree with them), so I will accept them for now.